### PR TITLE
#491 ReadableByteChannel in org.takes.rq.RqMultipart.Base never closes

### LINKED ISSUE #491: ReadableByteChannel in org.takes.rq.RqMultipart.Base never closes
When the ```private final transient ReadableByteChannel body;``` is initialized in the constructor 
as ```this.body = Channels.newChannel(stream);```, it is never closed anywhere in the class.

Try to close it in the method ```org.takes.rq.RqMultipart.Base.copy(WritableByteChannel, byte[])``` as:
```java
//The number of bytes read, possibly zero, or -1 if the channel has reached end-of-stream
if (this.body.read(this.buffer) == -1) {
    this.body.close();
    break;
}
```

### PR DESCRIPTION
**Pull request by xupyprmv**
#491. I added `.close()` call in finally block. Also I removed two checkstyle annotations that are not actual.

### CODE REVIEW

**Comment by davvd** [comment]
@xupyprmv Thanks, let me find someone who can review this pull request

**Comment by davvd** [comment]
@dmzaytsev please review, thanks

**Comment by dmzaytsev** [comment]
@xupyprmv looks good, thanks

**Comment by dmzaytsev** [comment]
@rultor merge

File: src/main/java/org/takes/rq/RqMultipart.java [hunk 1]
```diff
@@ -132,12 +130,20 @@
...
-            this.map = this.buildRequests(req);
+            try {
+                this.body = Channels.newChannel(stream);
+                this.buffer = ByteBuffer.allocate(
+                    // @checkstyle MagicNumberCheck (1 line)
+                    Math.min(8192, stream.available())
+                );
+                this.map = this.buildRequests(req);
+            } finally {
+                if (this.body == null) {
```
Comment (yegor256) [inline src/main/java/org/takes/rq/RqMultipart.java]: @xupyprmv `null` is an anti-pattern

**Comment by yegor256** [comment]
@xupyprmv what you're doing with this `if body == null` is a very bad approach. you need two try/catch blocks instead.

**Comment by xupyprmv** [comment]
@yegor256 I added changes according to your remark. Please, continue with CR.

**Comment by yegor256** [comment]
@xupyprmv looks correct now, but what about a unit test? How do you know that the code didn't work before your changes and works correctly after them?

**Comment by xupyprmv** [comment]
@yegor256 I added unit test. It not passes with old version and passes with new version.
I used not best approach (reflection) to create test, but it is better then open access to `body` field.

Please, continue with CR.

**Comment by xupyprmv** [comment]
@yegor256 Please, continue with CR.

**Comment by yegor256** [comment]
@xupyprmv oh, no... it's a terrible idea to do what you're doing in this test. the idea of a unit test is to help me to refactor my class without touching the test. your tests works in an opposite direction. never ever use reflection in unit testing. you're allowed to test only public interfaces of classes.

**Comment by xupyprmv** [comment]
@yegor256 

> looks correct now, but what about a unit test?

...

> oh, no... it's a terrible idea 

There is no other way to test private field without public access.

> you're allowed to test only public interfaces of classes 

So, I deleted test. Bug mentioned in #491 cannot be reproduced with test, that operates only with public interface. Also, fixed code cannot be approved by test, that operates only with public interface.

@yegor256 Please continue with CR.

**Comment by yegor256** [comment]
@rultor merge

**Comment by davvd** [comment]
@elenavolokhova please, check this issue for QA compliance, as per [par.24](http://at.teamed.io/policy.html#24)

**Comment by davvd** [comment]
@rultor deploy

**Comment by elenavolokhova** [comment]
@dmzaytsev 
According to our [quality rules](http://www.teamed.io/qa.html):
 > The code reviewer found at least three problems in the code.
 
 There were no issues found during review.
Please confirm that you will try to find at least three problems next time.

**Comment by dmzaytsev** [comment]
@elenavolokhova I confirm

**Comment by elenavolokhova** [comment]
@davvd Quality is acceptable here.

**Comment by davvd** [comment]
> @davvd Quality is acceptable here.

@elenavolokhova thank you, let's try to make it "good" next time

**Comment by davvd** [comment]
@dmzaytsev paied 10 mins to @elenavolokhova for QA review (payment ID is `75810218`)... **18 mins** added to your account (payment number `75810260`), many thanks for your contribution! 254 hours and 5 mins spent here.... extra minutes for review comments (c=3)... added +18 to your rating, now it is equal to [+2834](http://www.netbout.com/b/35166?open=rating)

**Pull request merged**